components/ide/jetbrains/backend-plugin/src/main/kotlin/io/gitpod/jetbrains/remote/GitpodCLIService.kt

```diff
@@ -46,9 +52,20 @@ class GitpodCLIService : RestService() {
             if (url.isNullOrBlank()) {
                 return "url is missing"
             }
-            return withClient(request, context) { project ->
-                BrowserUtil.browse(url, project)
+            GlobalScope.launch {
+                val controllerClientId = getControllerClientIdAsync()
+                runInEdt {
+                    ClientId.withClientId(controllerClientId) {
+                        BrowserUtil.browse(url)
+                    }
+                }
             }
+            sendOk(request, context)
```

Comment:
does it really work with open --wait?

Reply:
I think we should look into "gp open" in another PR, otherwise the "How to Test" section will become gigantic.

We also need another repo to test, cause Jenkins doesn't use "gp open" in their .gitpod.yml tasks.

(I suspect we don't need to change the "gp open" command as the file can be opened on the backend before the Thin Client connects. But we really need to investigate and confirm it.)

Reply:
There are only 3 scenarios:
- open
- open --wait
- preview

I don't think we need a special handling for each but one holistic solution about how to wait for a client.

Reply:
Got it.

I'll create a test-repo that has all those command in gitpod-tasks and use in place of Jenkins. Should also allow us to test the PR faster.

Reply:
To make sure you are trying to open files within project? 
There is a bug that it is not possible to open files outside of the project.

Reply:
Finally fixed the "gp open --wait" 🎉
Needed to wrap it with `ApplicationManager.getApplication().executeOnPooledThread`.